jmetal-core/src/main/java/org/uma/jmetal/operator/impl/crossover/PMXCrossover.java

```diff
@@ -109,20 +109,21 @@ public void setCrossoverProbability(double crossoverProbability) {
       // STEP 2: Get the subchains to interchange
       int replacement1[] = new int[permutationLength];
       int replacement2[] = new int[permutationLength];
-      for (int i = 0; i < permutationLength; i++)
-        replacement1[i] = replacement2[i] = -1;
+      subchainsinterchange(permutationLength, replacement1, replacement2);

```

Comment:
If you cut with explicitly named methods, then get rid of the comments. They add no value.

Reply:
However, I would argue that it is not necessary to create a function for 2 lines. I would rather merge the 2 arrays into a single matrix, integrate the instantiation of the matrix into the method which should then return the created matrix.